mentorship ios/Service/Networking/RelationAPI.swift

```diff
@@ -0,0 +1,108 @@
+    
+    func fetchTasks(id: Int, completion: @escaping ([TaskStructure], Bool) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        // make api call
+        tasksCancellable = NetworkManager.callAPI(urlString: URLStringConstants.MentorshipRelation.getCurrentTasks(id: id), token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(RelationModel().tasks) }
+            .sink {
+                var success = false
+                if NetworkManager.responseCode == 200 {
+                    success = true
+                }
+                completion($0, success)
+        }
+    }
+    
+    //create newtask api call
+    func addNewTask(newTask: RelationModel.AddTaskData, relationID: Int, completion: @escaping (RelationModel.ResponseData) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //prepare upload data
+        guard let uploadData = try? JSONEncoder().encode(newTask) else {
+            return
+        }
+        
+        //api call
+        cancellable = NetworkManager.callAPI(
+            urlString: URLStringConstants.MentorshipRelation.addNewTask(reqID: relationID),
+            httpMethod: "POST",
+            uploadData: uploadData,
+            token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(NetworkResponse(message: LocalizableStringConstants.networkErrorString)) }
+            .sink {
+                var success = false
+                if NetworkManager.responseCode == 201 {
+                    success = true
+                }
+                let response = RelationModel.ResponseData(message: $0.message, success: success)
+                completion(response)
+        }
+    }
+    
+    //mark task as complete api call
+    func markAsComplete(taskID: Int, relationID: Int, completion: @escaping (RelationModel.ResponseData) -> Void) {
+        //get auth token
+        guard let token = try? KeychainManager.getToken() else {
+            return
+        }
+        
+        //api call
+        cancellable = NetworkManager.callAPI(
+            urlString: URLStringConstants.MentorshipRelation.markAsComplete(reqID: relationID, taskID: taskID),
+            httpMethod: "PUT",
+            token: token)
+            .receive(on: RunLoop.main)
+            .catch { _ in Just(NetworkResponse(message: LocalizableStringConstants.networkErrorString)) }
+            .sink {
+                var success = false
+                if NetworkManager.responseCode == 200 {
+                    success = true
+                }
+                let response = RelationModel.ResponseData(message: $0.message, success: success)
+                completion(response)
+        }
+    }
+}
+
+// MARK: Network Model
+extension RelationAPI {
+    struct NetworkResponse: Decodable {
```

Comment:
integral part of service no need to keep it in extension

Reply:
Done